sqlmesh/core/test/runner.py

```diff
@@ -132,6 +132,7 @@ def _run_single_test(
             path=metadata.path,
             default_catalog=default_catalog,
             preserve_fixtures=preserve_fixtures,
+            lock=lock if num_workers > 1 else None,
```

Comment:
And we're reusing a lock that had a completely different purpose originally. Double red flag